src/components/Approval/Approval.tsx

```diff
@@ -0,0 +1,173 @@
+import { Button, Divider, Flex, Image, Link, SkeletonCircle } from '@chakra-ui/react'
+import { ChainTypes, SwapperType } from '@shapeshiftoss/types'
+import { useEffect, useRef, useState } from 'react'
+import { CountdownCircleTimer } from 'react-countdown-circle-timer'
+import { useFormContext } from 'react-hook-form'
+import { useHistory, useLocation } from 'react-router-dom'
+import { MiddleEllipsis } from 'components/MiddleEllipsis/MiddleEllipsis'
+import { Row } from 'components/Row/Row'
+import { SlideTransition } from 'components/SlideTransition'
+import { RawText, Text } from 'components/Text'
+import { useSwapper } from 'components/Trade/hooks/useSwapper/useSwapper'
+import { TradeState } from 'components/Trade/Trade'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import { useLocaleFormatter } from 'hooks/useLocaleFormatter/useLocaleFormatter'
+import { bn } from 'lib/bignumber/bignumber'
+import { theme } from 'theme/theme'
+
+type ApprovalParams = {
+  ethFiatRate: string
+}
+
+export const Approval = () => {
+  const history = useHistory()
+  const location = useLocation<ApprovalParams>()
+  const approvalInterval: { current: NodeJS.Timeout | undefined } = useRef()
+  const [approvalTxId, setApprovalTxId] = useState<string>()
+  const { ethFiatRate } = location.state
+
+  const {
+    getValues,
+    handleSubmit,
+    formState: { errors, isSubmitting }
+  } = useFormContext<TradeState<ChainTypes, SwapperType>>()
+  const { approveInfinite, checkApprovalNeeded, buildQuoteTx } = useSwapper()
+  const {
+    number: { toCrypto, toFiat }
+  } = useLocaleFormatter({ fiatType: 'USD' })
+  const {
+    state: { wallet }
+  } = useWallet()
+  const { quote, sellAsset, fees } = getValues()
+  const fee = fees?.chainSpecific?.approvalFee
+  const symbol = sellAsset.currency?.symbol
+
+  const approve = async () => {
+    if (!wallet) return
+    const txId = await approveInfinite(wallet)
+    if (!txId) return
+    setApprovalTxId(txId)
+    const interval = setInterval(async () => {
+      const approvalNeeded = await checkApprovalNeeded(wallet)
+      if (approvalNeeded) return
+      clearInterval(approvalInterval.current as NodeJS.Timeout)
```

Comment:
@0xdef1cafe still having issues with this one

Reply:
https://github.com/shapeshift/web/pull/264/commits/be0264f36b7a1c56af5a5fbdb8b1c3c99317baa1